src/sentry/tasks/files.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import absolute_import
+
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.deletion import MAX_RETRIES
+
+
+@instrumented_task(
+    name='sentry.tasks.files.delete_file',
+    queue='files',
+    default_retry_delay=60 * 5,
+    max_retries=MAX_RETRIES
+)
+def delete_file(path, **kwargs):
+    from sentry.models.file import get_storage
+    get_storage().delete(path)
```

Comment:
Could this raise an exception?

If so, is there an exception that w'd like to gracefully ignore? Like, if the file has already been deleted or can't be found, it'd be good to just silently ignore. Otherwise, the task will just keep retrying `MAX_RETRIES` times.

Reply:
`gcs_bucket.delete_blob(name)` will raise a `google.cloud.exceptions.NotFound`, and `response = s3_bucket.delete_objects(Delete={'Objects':[{'Key': name}]})` doesn't raise anything and instead you'll need to check like this: `err = response.get('Errors', None) ; if err['some error that means notfound or already deleted']:`

So to achieve that, we'll probably have to define custom exceptions (e.g. `DeletionSkipException`) inside the storage objects returned by `get_storage`, to catch and ignore. It'll be thrown by the delete method. How does that sound?

Reply:
Er, actually that is if you're interacting directly against the buckets `google.cloud.storage.Client.bucket('bucket-name')` and `boto3.resource('s3').Bucket('bucket-name')`. I actually haven't tried using the `delete` methods from our Storage classes.